optimum/intel/openvino/modeling_visual_language.py

```diff
@@ -601,17 +652,14 @@ def half(self):
         """
         Converts all the model weights to FP16 for more efficient inference on GPU.
         """
-        apply_moc_transformations(self.lm_model, cf=False)
-        compress_model_transformation(self.lm_model)
-        apply_moc_transformations(self.text_embeddings_model, cf=False)
-        compress_model_transformation(self.text_embeddings_model)
-        apply_moc_transformations(self.vision_embeddings_model, cf=False)
-        compress_model_transformation(self.vision_embeddings_model)
-        for part in self.additional_parts:
-            model = getattr(self, f"{part}_model", None)
-            if model is not None:
-                apply_moc_transformations(model, cf=False)
-                compress_model_transformation(model)
+        for _, submodel in self.submodels.items():
+            apply_moc_transformations(submodel, cf=False)
+            compress_model_transformation(submodel)
+        return self
+
+    def to(self, device):
+        self.language_model.to(device)
+        super().to(device)
```

Comment:
Missed this during previous review round. Is it on purpose that only language model device is changed?

Reply:
language model required special behaviour for setting the device as it contains 2 models and in some cases, potentially, it is possible to use it independently, other parts inherit device from base model and their update converted bysuper().to (change device in base model and call clear_requests

Reply:
`device_map`?

Reply:
Anyway, I would recommend doing this in the follow-up PR.